core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -776,7 +782,9 @@ public boolean doCheckExistingUser(String userName) throws UserStoreException {
                         if (userDN != null && userDN.length() > 0) {
                             bFound = true;
                             LdapName ldapName = new LdapName(userDN);
-                            userCache.put(userName, ldapName);
+                            if (userName != null) {
```

Comment:
It may be good to validate username once and then perform further checking instead of validating it again and again, WDYT ?